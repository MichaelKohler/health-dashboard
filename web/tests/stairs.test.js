/* global it, expect */
/* eslint-disable react/jsx-filename-extension */

import React from 'react';
import { useSelector } from 'react-redux';
import { shallowToJson  } from 'enzyme-to-json';
import { shallow } from 'enzyme';

import Stairs from '../src/stairs.jsx';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

it('renders correctly', () => {
  useSelector.mockImplementation(
    (selectorFn) => selectorFn({
      stairs: [{
        createdAt: '2019-08-07T08:00:03Z',
        stairs: 7,
      }],
      isFetchingHealth: false,
      failedFetchingHealth: false,
    }),
  );

  const wrapper = shallow(<Stairs />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('shows fetching paper', () => {
  useSelector.mockImplementation(
    (selectorFn) => selectorFn({
      stairs: [],
      isFetchingHealth: true,
      failedFetchingHealth: false,
    }),
  );

  const wrapper = shallow(<Stairs />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

/* eslint-enable react/jsx-filename-extension */
