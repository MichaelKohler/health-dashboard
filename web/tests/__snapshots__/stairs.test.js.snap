// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section>
  <div
    className="makeStyles-appBarSpacer-1"
  />
  <WithStyles(ForwardRef(Container))
    className="makeStyles-container-2"
    maxWidth="lg"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={3}
    >
      <StairsChart />
      <WithStyles(ForwardRef(Grid))
        item={true}
        lg={12}
        md={12}
        xs={12}
      >
        <StairsTable
          rows={
            Array [
              Object {
                "createdAt": "2019-08-07T08:00:03Z",
                "stairs": 7,
              },
            ]
          }
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Fab))
      aria-label="Add"
      className="makeStyles-fab-4"
      color="primary"
      onClick={[Function]}
    >
      <AddIcon />
    </WithStyles(ForwardRef(Fab))>
  </WithStyles(ForwardRef(Container))>
</section>
`;

exports[`shows fetching paper 1`] = `
<section>
  <div
    className="makeStyles-appBarSpacer-1"
  />
  <WithStyles(ForwardRef(Container))
    className="makeStyles-container-2"
    maxWidth="lg"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={3}
    >
      <StairsChart />
      <WithStyles(ForwardRef(Grid))
        item={true}
        lg={12}
        md={12}
        xs={12}
      >
        <WithStyles(ForwardRef(Paper))
          className="makeStyles-paper-3"
        >
          <p>
            Fetching...
          </p>
          <WithStyles(ForwardRef(CircularProgress))
            size={15}
          />
        </WithStyles(ForwardRef(Paper))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Fab))
      aria-label="Add"
      className="makeStyles-fab-4"
      color="primary"
      onClick={[Function]}
    >
      <AddIcon />
    </WithStyles(ForwardRef(Fab))>
  </WithStyles(ForwardRef(Container))>
</section>
`;
